cli/command/system/inspect.go

```diff
@@ -105,30 +105,51 @@ func inspectAll(ctx context.Context, dockerCli *command.DockerCli, getSize bool,
 	var inspectAutodetect = []struct {
 		ObjectType      string
 		IsSizeSupported bool
+		IsSwarmObject   bool
 		ObjectInspector func(string) (interface{}, []byte, error)
 	}{
-		{"container", true, inspectContainers(ctx, dockerCli, getSize)},
-		{"image", false, inspectImages(ctx, dockerCli)},
-		{"network", false, inspectNetwork(ctx, dockerCli)},
-		{"volume", false, inspectVolume(ctx, dockerCli)},
-		{"service", false, inspectService(ctx, dockerCli)},
-		{"task", false, inspectTasks(ctx, dockerCli)},
-		{"node", false, inspectNode(ctx, dockerCli)},
-		{"plugin", false, inspectPlugin(ctx, dockerCli)},
+		{"container", true, false, inspectContainers(ctx, dockerCli, getSize)},
+		{"image", false, false, inspectImages(ctx, dockerCli)},
+		{"network", false, false, inspectNetwork(ctx, dockerCli)},
+		{"volume", false, false, inspectVolume(ctx, dockerCli)},
+		{"service", false, true, inspectService(ctx, dockerCli)},
+		{"task", false, true, inspectTasks(ctx, dockerCli)},
+		{"node", false, true, inspectNode(ctx, dockerCli)},
+		{"plugin", false, false, inspectPlugin(ctx, dockerCli)},
 	}
 
+	// FIXME checking for literal string will be problematic if this error
+	// changes in a future release, and client is connecting to a daemon with a
+	// different version.
 	isErrNotSwarmManager := func(err error) bool {
-		return strings.Contains(err.Error(), "This node is not a swarm manager")
+		return strings.Contains(err.Error(), "is not a swarm manager")
+	}
+
+	// FIXME checking for literal string will be problematic if this error
+	// changes in a future release, and client is connecting to a daemon with a
+	// different version.
```

Comment:
I agree it's problematic. In addition to these concerns, there may be other errors we aren't considering which shouldn't be fatal for `inspect`.

I wonder if it would be better to ignore all errors returned from swarm-related endpoints. If there's really some problem communicating with the daemon, the interactions with non-swarm endpoints will trigger a visible error.